gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/SubscriptionKeysResource.java

```diff
@@ -75,7 +75,7 @@ public Response revokeKeySubscription(@PathParam("subscriptionId") String subscr
             hasPermission(RolePermission.API_SUBSCRIPTION, subscriptionEntity.getApi(), RolePermissionAction.UPDATE)
         ) {
             ApiKeyEntity apiKeyEntity = apiKeyService.findByKeyAndApi(apiKey, subscriptionEntity.getApi());
-            if (apiKeyEntity.getSubscription() != null && !subscriptionId.equals(apiKeyEntity.getSubscription())) {
+            if (apiKeyEntity.getSubscriptions().stream().map(SubscriptionEntity::getId).noneMatch(subscriptionId::equals)) {
```

Comment:
As SubscriptionEntity.equals() compares id ; could we simplify this expression using `apiKeyEntity.getSubscriptions().stream().noneMatch(subscriptionEntity)`

Or even, use `!apiKeyEntity.hasSubscription(subscriptionId)` ?